pkg/system/autoscaler.go

```diff
@@ -493,22 +496,23 @@ func (r *Reconciler) reconcilePrometheusAdapterResources(prometheus *monitoringv
 	if !util.KubeCreateSkipExisting(adapterAPIService) {
 		return fmt.Errorf("Error while crateiing APIService prometheus-adapter")
 	}
-	adapterHPA := util.KubeObject(bundle.File_deploy_internal_hpav2_autoscaling_yaml).(*autoscalingv2.HorizontalPodAutoscaler)
+	if err := r.ReconcileObject(r.AdapterHPA, r.reconcileAdapterHPA); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *Reconciler) reconcileAdapterHPA() error {
 	endpointsSpec := r.NooBaa.Spec.Endpoints
 	var minReplicas int32 = 1
 	var maxReplicas int32 = 2
 	if endpointsSpec != nil {
-		adapterHPA.Spec.MinReplicas = &endpointsSpec.MinCount
-		adapterHPA.Spec.MaxReplicas = endpointsSpec.MaxCount
-	} else {
-		adapterHPA.Spec.MinReplicas = &minReplicas
-		adapterHPA.Spec.MaxReplicas = maxReplicas
-	}
-	adapterHPA.Namespace = r.Request.Namespace
-	if err := r.ReconcileObject(adapterHPA, nil); err != nil {
-		return err
+		minReplicas = endpointsSpec.MinCount
+		maxReplicas = endpointsSpec.MaxCount
 	}
-
+	r.AdapterHPA.Spec.MinReplicas = &minReplicas
+	r.AdapterHPA.Spec.MaxReplicas = maxReplicas
+	r.AdapterHPA.Spec.Metrics[0].Object.Target.Value = resource.NewQuantity(1, resource.DecimalSI)
```

Comment:
Would you please add a comment about this line?

Reply:
@shirady I was getting an error when trying to update the HPA without `target.value`, So added this and it will not have any specific use since  `type`  is `AverageValue`.

Reply:
It can be easier to understand if you add a comment about the purpose of this line.

Reply:
I agree, let's explain this a bit... 
What do we expect as Metrics in place 0? shouldn't we build the whole spec?
In the file it hpav2-autoscaling.yaml it looks like we will create averageValue and not Value, I think they should look the same here and there

Reply:
Hi @shirady and @jackyalbo I managed to dive deep into this issue and find out that instead of setting a value I can set `nil` to this, the code is updated 
https://github.com/kubernetes/kubernetes/blob/afff019fbc90373127020693cdb056a968468f04/pkg/apis/autoscaling/validation/validation.go#L445C5-L445C7